game-core/src/test/java/games/strategy/triplea/delegate/battle/MustFightBattleExecutablesTest.java

```diff
@@ -0,0 +1,1670 @@
+
+    battle.setUnits(
+        List.of(defenderUnit), List.of(attackerUnit), List.of(), List.of(), defender, List.of());
+    final List<IExecutable> execs = battle.getBattleExecutables(true);
+
+    return Tuple.of(battle, execs);
+  }
+
+  private enum FirstStrikeBattleStep {
+    ATTACKER,
+    DEFENDER,
+    STANDARD,
+  }
+
+  private void assertThatFirstStrikeStepOrder(
+      final Tuple<MustFightBattle, List<IExecutable>> battleTuple,
+      final List<FirstStrikeBattleStep> stepOrder) {
+    final List<IExecutable> execs = battleTuple.getSecond();
+
+    final EnumMap<FirstStrikeBattleStep, Integer> indices =
+        new EnumMap<>(FirstStrikeBattleStep.class);
+
+    indices.put(
+        FirstStrikeBattleStep.ATTACKER,
+        getIndex(execs, MustFightBattle.FirstStrikeAttackersFire.class));
+    indices.put(
+        FirstStrikeBattleStep.DEFENDER,
+        getIndex(execs, MustFightBattle.FirstStrikeDefendersFire.class));
+    indices.put(
+        FirstStrikeBattleStep.STANDARD,
+        getIndex(execs, MustFightBattle.StandardAttackersFire.class));
+
+    assertThat(indices.get(stepOrder.get(0)), lessThan(indices.get(stepOrder.get(1))));
+    assertThat(indices.get(stepOrder.get(1)), lessThan(indices.get(stepOrder.get(2))));
+  }
+
+  private void assertThatFirstStrikeReturnFireIs(
+      final Tuple<MustFightBattle, List<IExecutable>> battleTuple,
+      final MustFightBattle.ReturnFire returnFire,
+      final boolean attacker) {
+    final MustFightBattle battle = battleTuple.getFirst();
+    final List<IExecutable> execs = battleTuple.getSecond();
+    final int index =
+        getIndex(
+            execs,
+            attacker
+                ? MustFightBattle.FirstStrikeAttackersFire.class
+                : MustFightBattle.FirstStrikeDefendersFire.class);
+    final IExecutable step = execs.get(index);
+
+    final IDelegateBridge delegateBridge = mock(IDelegateBridge.class);
+    step.execute(null, delegateBridge);
+
+    if (attacker) {
+      verify(battle).firstStrikeAttackersFire(returnFire);
+    } else {
+      verify(battle).firstStrikeDefendersFire(returnFire);
+    }
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is true, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is either, then attacker has return fire all")
+  void firstStrikeAttackerReturnFireAttHasDestroyerDefHasDestroyerWW2v2TrueSneakAttackTrueFalse() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, true, true, true, true),
+        MustFightBattle.ReturnFire.ALL,
+        true);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is true, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is either, then defender has return fire all")
+  void firstStrikeDefenderReturnFireAttHasDestroyerDefHasDestroyerWW2v2TrueSneakAttackTrueFalse() {
```

Comment:
Good use of the display name, though the test method name I think maybe became a bit too formulaic.

For example: `SneakAttackTrueFalse` does not really mean that much. I'd recommend looking at the method names and trying to state things more directly.  IE: The morpheus principle of naming, "Stop trying to hit me and hit me!"

Suggestion:
```
ww2v2DefenderHasReturnFireAllWhenBothAttackerAndDefenderHaveDestroyers()
```